zproject/backends.py

```diff
@@ -632,13 +632,32 @@ def is_account_control_disabled_user(self, ldap_user: _LDAPUser) -> bool:
         return ldap_disabled
 
     def is_account_realm_access_forbidden(self, ldap_user: _LDAPUser, realm: Realm) -> bool:
-        if "org_membership" not in settings.AUTH_LDAP_USER_ATTR_MAP:
+        if "org_membership" in settings.AUTH_LDAP_USER_ATTR_MAP:
+            org_membership_attr = settings.AUTH_LDAP_USER_ATTR_MAP["org_membership"]
+            allowed_orgs: List[str] = ldap_user.attrs.get(org_membership_attr, [])
+            return not is_subdomain_in_allowed_subdomains_list(realm.subdomain, allowed_orgs)
```

Comment:
If we want to make the two options not confusingly incompatible, which seems wise, I think this should be just 
```
if  not is_subdomain_in_allowed_subdomains_list(realm.subdomain, allowed_orgs):
    return True
```

and then fall through from there to check the second one.